dpnp/fft/dpnp_utils_fft.py

```diff
@@ -315,28 +369,39 @@ def _extract_axes_chunk(a, s, chunk_size=3):
     return a_chunks[::-1], s_chunks[::-1]
 
 
-def _fft(a, norm, out, forward, in_place, c2c, axes=None):
+def _fft(a, norm, out, forward, in_place, c2c, axes, batch_fft=True):
     """Calculates FFT of the input array along the specified axes."""
 
     index = 0
-    if axes is not None:  # batch_fft
-        len_axes = 1 if isinstance(axes, int) else len(axes)
+    fft_1d = isinstance(axes, int)
+    if batch_fft:
+        len_axes = 1 if fft_1d else len(axes)
         local_axes = numpy.arange(-len_axes, 0)
         a = dpnp.moveaxis(a, axes, local_axes)
         a_shape_orig = a.shape
         local_shape = (-1,) + a_shape_orig[-len_axes:]
         a = dpnp.reshape(a, local_shape)
         index = 1
+    else:
+        if not (c2c or fft_1d):
+            # for N-D real FFT, axes[-1] of a should be at the end
+            a = dpnp.moveaxis(a, axes[-1], -1)
```

Comment:
please add a test to cover that

Reply:
This was part of an earlier algorithm that I had implemented. It should be deleted.